agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -618,3 +618,197 @@ func TestNetworkModeBridge(t *testing.T) {
 	err := networkModeTest(t, agent, "bridge")
 	require.NoError(t, err, "Networking mode bridge testing failed")
 }
+
+// TestCustomAttributesWithValidOptions tests the ECS_INSTANCE_ATTRIBUTES
+// upon agent registration with a key, value pair
+func TestCustomAttributesWithValidOptions(t *testing.T) {
+	len := 25
+	key, value := GetRandomString(len), GetRandomString(len)
+
+	customAttribute := fmt.Sprintf("{\"%s\": \"%s\"}", key, value)
```

Comment:
A problem with random strings is that it introduces potential to fail non-deterministically.  In practice, an engineer encountering that will likely re-run the tests instead of investigating the error. 
I really like the idea of trying to find unique edge cases, but I don't think functional tests are the right medium for that approach.